addons/website/models/website_page.py

```diff
@@ -144,41 +145,8 @@ def unlink(self):
         return super(Page, self).unlink()
 
     def write(self, vals):
-        for page in self:
-            website_id = False
-            if vals.get('website_id') or page.website_id:
-                website_id = vals.get('website_id') or page.website_id.id
-
-            # If URL has been edited, slug it
-            if 'url' in vals:
-                url = vals['url']
-                redirect_old_url = redirect_type = None
-                # TODO This should be done another way after the backend/frontend merge
-                if isinstance(url, dict):
-                    redirect_old_url = url.get('redirect_old_url')
-                    redirect_type = url.get('redirect_type')
-                    url = url.get('url')
-                if not url.startswith('/'):
-                    url = '/' + url
-                if page.url != url:
-                    url = self.get_valid_page_url(url, website_id)
-                    page.menu_ids.write({'url': url})
-                    if redirect_old_url:
-                        self.env['website.rewrite'].create({
-                            'name': vals.get('name') or page.name,
-                            'redirect_type': redirect_type,
-                            'url_from': page.url,
-                            'url_to': url,
-                            'website_id': website_id,
-                        })
-                vals['url'] = url
-
-            # If name has changed, check for key uniqueness
-            if 'name' in vals and page.name != vals['name']:
-                vals['key'] = self.env['website'].with_context(website_id=website_id).get_unique_key(slugify(vals['name']))
-            if 'visibility' in vals:
-                if vals['visibility'] != 'restricted_group':
-                    vals['groups_id'] = False
```

Comment:
Hm.
It means that you could mess with the pages now in the page form view, which is accessible through the ir.ui.view for instance (there is a field/link to its page).
Doesn't it work to keep it here? Since the wizard is calling write on the page anyway, it will go through here?

Reply:
Indeed, this should be on `website.page` to handle both cases... I tweaked the code now to make it work this way :+1: